src/tribler-core/tribler_core/modules/bandwidth_accounting/database.py

```diff
@@ -87,6 +87,16 @@ def get_latest_transaction(self, public_key_a: bytes, public_key_b: bytes) -> Ba
         db_obj = self.BandwidthTransaction.get(public_key_a=public_key_a, public_key_b=public_key_b)
         return BandwidthTransactionData.from_db(db_obj) if db_obj else None
 
+    @db_session
+    def get_latest_transactions(self, public_key: bytes) -> List[BandwidthTransactionData]:
+        """
+        Return the latest transactions of a given public key, or an empty list if no transactions exist.
+        :param public_key: The public key of the party transferring the bandwidth.
+        :return The latest transactions of the specified public key, or an empty list if no transactions exist.
+        """
+        db_objs = self.BandwidthTransaction.select(lambda bt: public_key in (bt.public_key_a, bt.public_key_b))[:]
```

Comment:
Is the `[:]` necessary here?

Seems like `db_objs` will be used only for iterating.

Reply:
Did you check this request on large databases? Transaction's DB can be large...

Reply:
- Good point. `[:]` is indeed not necessary.
- The number of transactions is small compared to previous Trustchain database unless there are thousands of nodes but a good point to add a limit. I'll make the change.